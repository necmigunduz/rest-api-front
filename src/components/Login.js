import { Redirect, useLocation } from 'react-router-dom';
import { useState } from 'react';
import { login } from '../sessions/sessions';
import { saveToken } from '../localStorage/storage';

const Login = () => {
  const [redirect, setRedirect] = useState();
  const [error, setError] = useState();

  const location = useLocation();
  const path = location.pathname.split('/')[2];

  const signUp = path !== 'login';
  const submitValue = signUp ? 'Sign Up' : 'Login';
  const endpoint = signUp ? 'users' : 'authentication';
  const validatePassword = signUp ? <input className="field m-b-20 background-blue color-white" id="rPassword" type="password" placeholder="reapeat password" /> : <div />;
  const link = signUp ? <a href="/users/login">Login</a> : <a href="/users/sign-up">Sign Up</a>;

  const handleSubmit = async (e) => {
    e.preventDefault();
    const name = e.target.parentElement[0].value;
    const password = e.target.parentElement[1].value;
    const rPassword = e.target.parentElement[2].value;
    if (signUp && password !== rPassword) return;
    const response = await login(name, password, endpoint);
    if (response.token) {
      saveToken(response.token);
      setRedirect(<Redirect to={{ pathname: '/' }} />);
    } else {
      setError(<div className="m-b-20">{response.message}</div>);
    }
  };

  return (
    <div className="login text-center">
      {redirect}
      <span>{error}</span>
      <form>
        <h6 className="m-b-10 color-gray">Enter your username</h6>
        <input className="field m-b-20 background-blue color-white" id="name" type="text" placeholder="username" />
        <h6 className="m-b-10 color-gray">Enter your password</h6>
        <input className="field m-b-20 background-blue color-white" id="password" type="password" placeholder="password" />
        {validatePassword}
        <input className="login-btn color-white background-dark-blue m-b-30" type="submit" value={submitValue} onClick={(e) => handleSubmit(e)} />
      </form>
      <p>or</p>
      {link}
    </div>
  );
};

export default Login;
