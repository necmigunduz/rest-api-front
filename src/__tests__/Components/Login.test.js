import renderer from 'react-test-renderer';
import Login from '../../components/Login';

jest.mock('react-router-dom', () => ({
  ...jest.requireActual('react-router-dom'),
  useLocation: () => ({
    pathname: 'www.instance.com/test',
  }),
}));

describe('Login', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<Login />).toJSON();
    expect(tree).toMatchSnapshot();
  });
});
