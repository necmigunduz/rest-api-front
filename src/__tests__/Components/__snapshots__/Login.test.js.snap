// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Login renders correctly 1`] = `
<div
  className="login text-center"
>
  <form>
    <h6
      className="m-b-10 color-gray"
    >
      Enter your username
    </h6>
    <input
      className="field m-b-20 background-blue color-white"
      id="name"
      placeholder="username"
      type="text"
    />
    <h6
      className="m-b-10 color-gray"
    >
      Enter your password
    </h6>
    <input
      className="field m-b-20 background-blue color-white"
      id="password"
      placeholder="password"
      type="password"
    />
    <input
      className="field m-b-20 background-blue color-white"
      id="rPassword"
      placeholder="reapeat password"
      type="password"
    />
    <input
      className="login-btn color-white background-dark-blue m-b-30"
      onClick={[Function]}
      type="submit"
      value="Sign Up"
    />
  </form>
  <p>
    or
  </p>
  <a
    href="/users/login"
  >
    Login
  </a>
</div>
`;
